README.md

```diff
@@ -1,69 +1,77 @@
-may compute at run time.  It issues a warning if the program may TODO.
-It works via a technique called pluggable typechecking.
+At compile time, the Guice Checker estimates what bindings the program may compute at run time.  It issues a warning if the program attempts to request a binding that has not been properly defined or configured. It works via a technique called pluggable typechecking.
 
-You need to specify the contracts of methods and fields in your code --
-that is, their requirements and their guarantees.  The Dependency InjectionChecker
-ensures that your code is consistent with the contracts, and that the
-contracts guarantee that TODO.
+Most of the time, the Guice Checker will work automatically. However, sometimes you may need to write an annotation to specify dataflow that is relevant to Guice bindings, if it crosses procedure boundaries. You can achieve that with the following type annotations:
 
-You specify your code by writing *qualifiers* such as `@DependencyInjectionBottom`
-on types, to indicate more precisely what values the type represents.
-Here are the type qualifiers that are supported by the Dependency InjectionChecker:
+`@Bind({})`
 
-`@DependencyInjectionUnknown`:
-The value might or might not be TODO. It is not safe to use for TODO.
-This is the default type, so programmers usually do not need to write it.
+- The default qualifier of this system. If an expression's type has this qualifier, then the expression's value may not result from a direct method invocation of `com.google.inject.AbstractModule.bind`. It implies that no arguments have been passed to the `bind` method. Since this is the default type, programmers usually do not need to write it.
 
-`@DependencyInjectionBottom`:
-The value is definitely TODO. It is safe to use for TODO.
+`@Bind({Baz.class})`
 
+- If an expression's type has this qualifier, then the expression's value was definitely the result of a direct method invocation of `com.google.inject.AbstractModule.bind` that received the argument `Baz.class` in an `AbstractModule`. It's value represents the argument that was passed to the invocation of the `bind` method.
 
-## How to build the checker
+`@BindAnnotatedWith({Baz.class}, {"someName"})`
 
-Run these commands from the top-level directory.
+- If an expression's type has this qualifier, then the expression is definitely a call to `com.google.inject.binder.AnnotatedBindingBuilder.annotatedWith` on an `AnnotatedBindingBuilder` that received the argument `Baz.class` in an `AbstractModule`. It represents two values, the first one being the argument that was passed to the invocation of the `bind` method and the second being the argument that was passed to the invocation of the `annotatedWith` method. It is important to note that this qualifier is always a _supertype_ of `@Bind({Baz.class})` and a _subtype_ of `@Bind({})`. It can only ever contain the result of an `@Bind`, thus it cannot exist on its own.
```

Comment:
I'd use a different word than "values" for the two things that this represents, because "value" has a technical term when discussing compilers that is different than what you mean here. To be honest, "things" might be sufficient.